### PR TITLE
handle APEv2 frames by skipping them

### LINKED ISSUE #6: Handling APE tags
Hi! 

Thanks for creating this library and doing it properly (e.g. not "estimating" the length but really measuring it)!

There were some files in my collection which tripped up however, with the error:
`Unexpected frame, header: 1095779668 `
In hex thats 0x41504554, which can be read as APET. This points to APE-Tags in the file (https://en.wikipedia.org/wiki/APE_tag). Until today I didn't even know that these could be inside of mp3 files, but I would really appreciate it if you could add some code to skip these.

### PR DESCRIPTION
**Pull request by mqus**
Fixes #6

### CODE REVIEW

**Comment by mqus** [comment]
I only check for the start of the APETAGEX message, but that should be enough for now. I tested this with my files locally but as I'm probably not allowed to upload ("distribute") them, I held back on including it here.

File: src/lib.rs [hunk 1]
```diff
@@ -314,6 +314,31 @@ where
             continue;
         }
 
+        // APEv2 frame
+        let is_ape_v2 = header_buffer[0] == 'A' as u8
+            && header_buffer[1] == 'P' as u8
+            && header_buffer[2] == 'E' as u8
+            && header_buffer[3] == 'T' as u8;
```
Comment (agersant) [inline src/lib.rs:321]: Can you check header_buffer[4] through header_buffer [7] for the full preamble (http://wiki.hydrogenaud.io/index.php?title=APE_Tags_Header)?

**Review by agersant — CHANGES_REQUESTED** [review]
Can you also add a unit test? You can use this file (same as the other test files in the repo) for validation:
http://www.antoinegersant.com/tmp/APEv2.mp3

File: src/lib.rs (hunk 1 above)
Comment (mqus) [reply src/lib.rs:321]: header_buffer is only 4 bytes big,
I'll use ape_header instead and throw an UnexpectedFrame Error if it's wrong, would that be ok?

Comment (agersant) [reply src/lib.rs:321]: That works for me!

**Review by agersant — APPROVED** [review]

**Pull request merged**